src/executor/query/DedupExecutor.cpp

```diff
@@ -15,31 +15,29 @@ folly::Future<Status> DedupExecutor::execute() {
     SCOPED_TIMER(&execTime_);
     auto* dedup = asNode<Dedup>(node());
     DCHECK(!dedup->inputVar().empty());
-    auto iter = ectx_->getResult(dedup->inputVar()).iter();
+    Result result = ectx_->getResult(dedup->inputVar());
 
-    if (UNLIKELY(iter == nullptr)) {
+    if (UNLIKELY(result.iterRef() == nullptr)) {
```

Comment:
I don't think this is a good practice. Why not use the iterator directly, but call `result.iterRef` for several times?

Reply:
Some result will reference the value like `filter`, so we could copy the origin result and save it.

Reply:
I have some thought about the `iterRef` interface. 

firstly, we should consider why we define a `iter` function which always returns its copy. If we must let user to access the inner data, why to return the shared_ptr rather than unique_ptr? It maybe represent that we don't do well designs if we need to provide so many functions such as  `iter` and `iterRef`.

Reply:
This is only temporary method. I think we just provide method `iter`, and user need manually call `copy` if need a copy.

Reply:
I don't modify it completely for it's not related to this PR, will fix it later.